server/channels/api4/user_test.go

```diff
@@ -7917,3 +7919,559 @@ func TestLoginWithDesktopToken(t *testing.T) {
+	})
+
+	t.Run("Get users without permissions", func(t *testing.T) {
+		th.Client.Logout(context.Background())
+		_, resp, err := th.Client.GetUsersByUsernames(context.Background(), []string{th.BasicUser.Username})
+		require.Error(t, err)
+		CheckUnauthorizedStatus(t, resp)
+	})
+
+	t.Run("Get users as system admin", func(t *testing.T) {
+		users, _, err := th.SystemAdminClient.GetUsersByUsernames(context.Background(), []string{th.BasicUser.Username})
+		require.NoError(t, err)
+		require.Len(t, users, 1)
+		require.Equal(t, th.BasicUser.Username, users[0].Username)
+		CheckUserSanitization(t, users[0])
+	})
+}
+
+func TestGetFilteredUsersStats(t *testing.T) {
+    th := Setup(t).InitBasic()
+    defer th.TearDown()
+
+    t.Run("Get filtered users stats as system admin", func(t *testing.T) {
+        options := &model.UserCountOptions{
+            TeamId: th.BasicTeam.Id,
+            IncludeDeleted: false,
+            IncludeBotAccounts: false,
+            IncludeRemoteUsers: true,
+        }
+        stats, resp, err := th.SystemAdminClient.GetFilteredUsersStats(context.Background(), options)
+        require.NoError(t, err)
+        CheckOKStatus(t, resp)
+        require.NotNil(t, stats)
+        require.True(t, stats.TotalUsersCount > 0)
+    })
+
+    t.Run("Get filtered users stats as regular user", func(t *testing.T) {
+        options := &model.UserCountOptions{
+            TeamId: th.BasicTeam.Id,
+            IncludeDeleted: false,
+            IncludeBotAccounts: false,
+            IncludeRemoteUsers: true,
+        }
+        _, resp, err := th.Client.GetFilteredUsersStats(context.Background(), options)
+        require.Error(t, err)
+        CheckForbiddenStatus(t, resp)
+    })
+
+    t.Run("Get filtered users stats with invalid team id", func(t *testing.T) {
+		options := &model.UserCountOptions{
+			TeamId: "invalid_team_id",
+			IncludeDeleted: false,
+			IncludeBotAccounts: false,
+			IncludeRemoteUsers: true,
+		}
+		stats, resp, err := th.SystemAdminClient.GetFilteredUsersStats(context.Background(), options)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+		require.NotNil(t, stats)
+		// The server seems to return stats even with an invalid team ID
+		// We should check that the returned stats make sense in this context
+		require.Equal(t, int64(0), stats.TotalUsersCount, "Expected 0 users for an invalid team ID")
+	})
+
+    t.Run("Get filtered users stats with roles", func(t *testing.T) {
+        options := &model.UserCountOptions{
+            TeamId: th.BasicTeam.Id,
+            IncludeDeleted: false,
+            IncludeBotAccounts: false,
+            IncludeRemoteUsers: true,
+            Roles: []string{model.SystemUserRoleId},
+        }
+        stats, resp, err := th.SystemAdminClient.GetFilteredUsersStats(context.Background(), options)
+        require.NoError(t, err)
+        CheckOKStatus(t, resp)
+        require.NotNil(t, stats)
+        require.True(t, stats.TotalUsersCount > 0)
```

Comment:
Better assertion would be get the number of users from `th.App` and then compare it from the client side.